c/utils/terminal/terminal.cc

```diff
@@ -143,12 +155,20 @@ bool Terminal::unicode_allowed() const noexcept {
 }
 
 int Terminal::get_width() {
+#if DT_OS_WINDOWS
+  _detect_window_size();
+#else
```

Comment:
One problem here is that if i needed both width&height, then `_detect_window_size()` would be called twice.

Possible solution is to reset the `width_` to 0 after returning it, then `height_` will remain with a proper value which can be queried (and autoreset) too. Something like
```C++
int Terminal::get_width() {
  if (!width_) _detect_window_size();
  int w = width_;
  #if DT_OS_WINDOWS
    width_ = 0;
  #endif
  return w;
}
```

Reply:
Calling `_detect_window_size()` is pretty cheap, but if we want to make it optimal I would just propose to create a single method called `get_size()`: it will return a structure with `.width` and `.height` fields. 

Btw, I don't see we're using `get_heigh()` anywhere in the code.

Reply:
Yeah, I guess `get_height()` isn't nearly as useful as it looks...